File: backend/app/schemas/medical_report.py
from typing import Literal

from pydantic import BaseModel, Field


RiskLevel = Literal["Low", "Moderate", "High"]


class MedicalReportAnalyzeRequest(BaseModel):
    report_text: str = Field(min_length=3, max_length=6000)


class MedicalReportAnalyzeResponse(BaseModel):
    summary: str
    important_findings: list[str]
    risk_level: RiskLevel
    recommended_next_step: str
    questions_to_ask_doctor: list[str]
    medical_disclaimer: str

File: frontend/lib/medical-report.ts
import { apiClient } from "@/lib/api-client";

export type MedicalReportAnalyzeRequest = {
  report_text: string;
};

export type MedicalReportAnalyzeResponse = {
  summary: string;
  important_findings: string[];
  risk_level: "Low" | "Moderate" | "High";
  recommended_next_step: string;
  questions_to_ask_doctor: string[];
  medical_disclaimer: string;
};

export async function analyzeMedicalReport(
  payload: MedicalReportAnalyzeRequest,
): Promise<MedicalReportAnalyzeResponse> {
  return apiClient<MedicalReportAnalyzeResponse>("/api/medical-report/analyze", {
    errorMessage: "Unable to analyze medical report",
    method: "POST",
    body: payload,
  });
}
